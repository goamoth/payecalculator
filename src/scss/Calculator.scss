 @import './Main.scss';

.cal-wrapper {
  border: 1px solid $border;
  background: $white;
  max-width: 750px;
  height: auto !important;
  margin: 30px auto;
  @include border-radius(5px, 5px, 5px, 5px);
}
