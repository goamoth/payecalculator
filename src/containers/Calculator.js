import React, { Component } from 'react';
import Header from '../components/Header';
import Form from '../components/Form';
import Results from '../components/Results';
import '../scss/Calculator.scss';

class Calculator extends Component {
   constructor(props) {
      super(props);
      this.state = {
         year: '',
         grossPay: 0,
         nssf: '',
         isExempted: 'false'
      };
      
   }

   handleChange = (e) => {
      let name = e.target.name;

      if (e.target.name === 'isExempted') {
         this.setState( (prevState) => {
            let isChecked = prevState.isExempted === 'false' ? 'true' : 'false';
            return {
               isExempted: isChecked
            }
         });

      } else {
         this.setState({
            [name]: e.target.value
         });
      }

   };

   onSubmit = (e) => {
      e.preventDefault();
      alert('Hola! Am now on the parent Component!');
   };

   render() {
      return (
         <div className="row">
            <div className="col-12 cal-wrapper">
               <Header headerType="h1" headerText="PAYE Calculator" />
               <Form {...this.state} changeHandler={this.handleChange} onSubmit={this.onSubmit} />
            </div>

            <div className="w-100"></div>

            <div className="col-12">
               <Results {...this.state} />
            </div>
         </div>
      );
   }
}

export default Calculator;