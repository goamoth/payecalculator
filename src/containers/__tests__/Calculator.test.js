import React from 'react';
import { shallow } from 'enzyme';
import Calculator from '../Calculator';
import renderer from 'react-test-renderer';

describe('Calculator', () => {
   let mountedCal;

   beforeEach(() => {
      mountedCal = shallow(<Calculator />);
   });

   it('renders without crashing', () => {
      expect(mountedCal).toBeDefined();
   });

   it('Renders correctly', () => {
      const tree = renderer.create(<Calculator />).toJSON();
      expect(tree).toMatchSnapshot();
   });

   it('Renders Header Component', () => {
      const mountedHeader = mountedCal.find('Header');
      expect(mountedHeader.length).toBe(1);
   });

   it('Renders Form component', () => {
      const mountedForm = mountedCal.find('Form');
      expect(mountedForm.length).toBe(1);
   });

   it('Renders Results component', () => {
      const mountedResults = mountedCal.find('Results');
      expect(mountedResults.length).toBe(1);
   });
});