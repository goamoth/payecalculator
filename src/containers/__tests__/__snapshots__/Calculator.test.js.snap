// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-12 cal-wrapper"
  >
    <div
      className="row header-wrapper"
    >
      <div
        className="col-12"
      >
        <h1
          className="normal-heading"
        >
          PAYE Calculator
        </h1>
      </div>
    </div>
    <div
      className="row form-wrapper"
    >
      <div
        className="col-12"
      >
        <form
          className="theForm"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="year"
            >
               Taxation Year: 
            </label>
            <select
              className="custom-select"
              id="year"
              name="year"
              onChange={[Function]}
              value=""
            >
              <option
                selected={true}
                value=""
              >
                 - Select Year - 
              </option>
              <option
                value="2020"
              >
                 2020 
              </option>
            </select>
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="grossPay"
            >
               Gross Income: 
            </label>
            <input
              className="form-control"
              id="grossPay"
              name="grossPay"
              onChange={[Function]}
              placeholder="75000"
              type="number"
              value={0}
            />
          </div>
          <div
            className="form-group"
          >
            <label
              style={
                Object {
                  "display": "block",
                }
              }
            >
               NSSF Contribution: 
            </label>
            <div
              className="custom-control custom-radio custom-control-inline"
            >
              <input
                checked={false}
                className="custom-control-input"
                id="old"
                name="nssf"
                onChange={[Function]}
                type="radio"
                value="old"
              />
              <label
                className="custom-control-label"
                htmlFor="old"
              >
                 Old (200) 
              </label>
            </div>
            <div
              className="custom-control custom-radio custom-control-inline"
            >
              <input
                checked={false}
                className="custom-control-input"
                id="new"
                name="nssf"
                onChange={[Function]}
                type="radio"
                value="new"
              />
              <label
                className="custom-control-label"
                htmlFor="new"
              >
                 New (Tiered) 
              </label>
            </div>
          </div>
          <div
            className="form-group disability"
          >
            <div
              className="custom-control custom-checkbox"
            >
              <input
                checked={false}
                className="custom-control-input"
                id="isExempted"
                name="isExempted"
                onChange={[Function]}
                type="checkbox"
                value="true"
              />
              <label
                className="custom-control-label"
                htmlFor="isExempted"
              >
                 I have Disability Tax Exemption Certificate 
              </label>
            </div>
          </div>
          <div
            className="form-group"
          >
            <button
              className="btn btn-primary"
              type="submit"
            >
              Calculate
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="w-100"
  />
  <div
    className="col-12"
  >
    <div
      className="row results-wrapper"
    >
      <div
        className="col-12"
      >
        <div
          className="row header-wrapper"
        >
          <div
            className="col-12"
          >
            <h3
              className="normal-heading"
            >
              Results
            </h3>
          </div>
        </div>
        <table
          className="table table-striped"
        >
          <tbody>
            <tr>
              <td>
                 Gross Income: 
              </td>
              <td>
                 Year: 
                
                 - GrossPay: 
                0
                 - NSSF: 
                
                 - isExempted: 
                false
                 
              </td>
            </tr>
            <tr>
              <td>
                 NSSF Contribution: 
              </td>
              <td>
                 0 
              </td>
            </tr>
            <tr>
              <td>
                 Total Deduction: 
              </td>
              <td>
                 0 
              </td>
            </tr>
            <tr>
              <td>
                 Taxable Income: 
              </td>
              <td>
                 0 
              </td>
            </tr>
            <tr>
              <td>
                 Personal Relief: 
              </td>
              <td>
                 0 
              </td>
            </tr>
            <tr>
              <td>
                 PAYE: 
              </td>
              <td>
                 0 
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
